3rdParty/velocypack/include/velocypack/HashedStringRef.h

```diff
@@ -0,0 +1,272 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2020 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Max Neunhoeffer
+/// @author Jan Steemann
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef VELOCYPACK_HASHED_STRINGREF_H
+#define VELOCYPACK_HASHED_STRINGREF_H 1
+
+#include <cstring>
+#include <functional>
+#include <algorithm>
+#include <string>
+#include <iosfwd>
+
+#include "velocypack/velocypack-common.h"
+
+namespace arangodb {
+namespace velocypack {
+class Slice;
+class StringRef;
+
+class HashedStringRef {
+ public:
+  /// @brief create an empty HashedStringRef
+  HashedStringRef() noexcept 
+    : _data(""), 
+      _length(0), 
+      _hash(hash("", 0)) {}
+
+  /// @brief create a HashedStringRef from an std::string
+  explicit HashedStringRef(std::string const& str) noexcept 
+    : HashedStringRef(str.data(), str.size()) {}
+  
+  /// @brief create a HashedStringRef from a string plus length
+  HashedStringRef(char const* data, std::size_t length) noexcept 
+    : _data(data), 
+      _length(static_cast<uint32_t>(length)), 
+      _hash(hash(_data, _length)) {}
+ 
+  /// @brief create a HashedStringRef from a string plus length and hash.
+  /// note: the hash is *not* validated. it is the caller's responsibility
+  /// to ensure the hash value is correct
+  HashedStringRef(char const* data, std::size_t length, uint32_t hash) noexcept 
+    : _data(data), 
+      _length(static_cast<uint32_t>(length)), 
```

Comment:
do you think it's safe enough to silently cast `size_t` to `uint32_t`?

Reply:
addressed